src/NuGet.Clients/NuGet.PackageManagement.UI/Models/DetailControlModel.cs

```diff
@@ -519,6 +524,59 @@ internal async Task LoadPackageMetadaAsync(IPackageMetadataProvider metadataProv
             }
         }
 
+        private async Task<IEnumerable<VersionInfoWithDeprecationMetadata>> GetVersionsWithDeprecationMetadataAsync(IPackageMetadataProvider metadataProvider)
+        {
+            var versions = await _searchResultPackage.GetVersionsAsync();
+            return await Task.WhenAll(
+                versions.Select(VersionInfoWithDeprecationMetadata.FromVersionInfoAsync));
+        }
+
+        private class VersionInfoWithDeprecationMetadata
```

Comment:
I couldn't decide between which approach was best or what the team preferred. I can switch to just tuples.

Reply:
I means [ValueTuple](https://blogs.msdn.microsoft.com/mazhou/2017/05/26/c-7-series-part-1-value-tuples/), not `Tuple<>`. `Tuple<>` is bad because everything becomes meaningless `.Item1`, `.Item2`, but ValueTuples with C#7 allow you to have meaningful names still.

Reply:
Oh wow, never used that before, will test it out.

Reply:
Done!